### ISSUE TITLE
The language server throws an exception when hovering over a variable's name

### ISSUE THREAD

**Issue by ionut-botizan** [felixfbecker/php-language-server#310]
In some cases, when hovering over a variable's name, the language server throws an exception which causes the `Output` panel to open (which is **very** annoying).

The message displayed in the `Output` panel is:
```
[Error - 10:32:21] Request textDocument/hover failed.
  Message: ErrorException: Undefined property: PhpParser\Node\Expr\AssignOp\Plus::$left in C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\DefinitionResolver.php:589
Stack trace:
#0 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\DefinitionResolver.php(589): {closure}(8, 'Undefined prope...', 'C:\\Users\\ionut....', 589, Array)
#1 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\DefinitionResolver.php(796): LanguageServer\DefinitionResolver->resolveExpressionNodeToType(Object(PhpParser\Node\Expr\AssignOp\Plus))
#2 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\DefinitionResolver.php(131): LanguageServer\DefinitionResolver->getTypeFromNode(Object(PhpParser\Node\Expr\AssignOp\Plus))
#3 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\DefinitionResolver.php(156): LanguageServer\DefinitionResolver->createDefinitionFromNode(Object(PhpParser\Node\Expr\AssignOp\Plus))
#4 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\src\Server\TextDocument.php(322): LanguageServer\DefinitionResolver->resolveReferenceNodeToDefinition(Object(PhpParser\Node\Expr\Variable))
#5 [internal function]: LanguageServer\Server\TextDocument->LanguageServer\Server\{closure}()
#6 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\coroutine.php(70): Generator->send(Object(LanguageServer\PhpDocument))
#7 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\Promise.php(242): Sabre\Event\{closure}(Object(LanguageServer\PhpDocument))
#8 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\Loop\Loop.php(261): Sabre\Event\Promise->Sabre\Event\{closure}()
#9 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\Loop\Loop.php(215): Sabre\Event\Loop\Loop->runNextTicks()
#10 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\Loop\Loop.php(194): Sabre\Event\Loop\Loop->tick(true)
#11 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\sabre\event\lib\Loop\functions.php(122): Sabre\Event\Loop\Loop->run()
#12 C:\Users\ionut.botizan\.vscode\extensions\felixfbecker.php-intellisense-1.1.2\vendor\felixfbecker\language-server\bin\php-language-server.php(50): Sabre\Event\Loop\run()
#13 {main}
  Code: -32603 
```

You should be able to replicate this (I did it on two different machines) by creating a PHP file and writing this piece of code:
```php
<?php

$params += ['from' => 'today'];
doSomethingWith($params['from']);

?>
```
Save then hover over the `$params` variable on line 4. This should trigger the error.

If the problem lies within the `PhpParser` library, can you at least please `catch` the exception and silently discard it so the `Output` panel wouldn't keep opening?

Best Regards,

Ionuț

**Comment by felixfbecker** [comment]
Probably fixed by #307

**Comment by ionut-botizan** [comment]
I just tried it with the current `master` branch  (56bd465) and it seems to be fixed. 👍
